File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>>res=new ArrayList<>();
        int [][] arr=new int[numRows][numRows];
        for(int i=0;i<numRows;i++){
            for(int j=0;j<i+1;j++){
                if(j==0 || j==i){
                    arr[i][j]=1;
                }
                else{
                    arr[i][j]=arr[i-1][j-1]+arr[i-1][j];
                }
            }
        }

        for(int i=0;i<arr.length;i++){
            List<Integer>lst=new ArrayList<>();
            for(int j=0;j<i+1;j++){
                lst.add(arr[i][j]);
            }
            res.add(lst);
        }
        return res;
        
    }
    
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res=[]
        arr=[[0]*numRows for _ in range(numRows)]



        for i in range(numRows):
            for j in range(i+1):
                if j==0 or j==i:
                    arr[i][j]=1
                else:
                    arr[i][j]=arr[i-1][j-1]+arr[i-1][j]

        for i in range(numRows):
            lst=[]
            for j in range(i+1):
                lst.append(arr[i][j])

            res.append(lst)
        

        return res
                    

        